### PR TITLE
Fix IDs

### LINKED ISSUE #1013: Crash with new Pleroma IDs on Nightly/Master
Trying to make a number out of string to give placeholder some value.

<details>

<summary>
Stacktrace
</summary>

On master:

```
    Process: com.keylesspalace.tusky, PID: 28097
    java.lang.NumberFormatException: Invalid BigInteger: 9fGYpSx9Qrt1QsrP5k
        at java.math.BigInt.invalidBigInteger(BigInt.java:86)
        at java.math.BigInt.checkString(BigInt.java:140)
        at java.math.BigInt.putDecString(BigInt.java:90)
        at java.math.BigInteger.<init>(BigInteger.java:217)
        at com.keylesspalace.tusky.fragment.TimelineFragment.onLoadMore(TimelineFragment.java:791)
        at com.keylesspalace.tusky.fragment.TimelineFragment.access$300(TimelineFragment.java:98)
        at com.keylesspalace.tusky.fragment.TimelineFragment$3.onLoadMore(TimelineFragment.java:443)
        at com.keylesspalace.tusky.view.EndlessOnScrollListener.onScrolled(EndlessOnScrollListener.java:45)
        at com.keylesspalace.tusky.fragment.TimelineFragment$3.onScrolled(TimelineFragment.java:423)
        at androidx.recyclerview.widget.RecyclerView.dispatchOnScrolled(RecyclerView.java:4961)
        at androidx.recyclerview.widget.RecyclerView.scrollByInternal(RecyclerView.java:1954)
        at androidx.recyclerview.widget.RecyclerView.onTouchEvent(RecyclerView.java:3187)
        at android.view.View.dispatchTouchEvent(View.java:11776)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2962)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2643)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2657)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2657)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2657)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2657)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2657)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2657)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2657)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2657)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2657)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2657)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2657)
        at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:448)
        at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1829)
        at android.app.Activity.dispatchTouchEvent(Activity.java:3307)
        at androidx.appcompat.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:69)
        at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:410)
        at android.view.View.dispatchPointerEvent(View.java:12015)
        at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:4795)
        at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4609)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4147)
        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4200)
        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4166)
        at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4293)
        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4174)
        at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4350)
2019-01-28 21:21:53.000 28097-28097/? E/AndroidRuntime:     at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4147)
        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4200)
        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4166)
        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4174)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4147)
        at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:6661)
        at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:6635)
        at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:6596)
        at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:6764)
        at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:186)
        at android.view.InputEventReceiver.nativeConsumeBatchedInputEvents(Native Method)
        at android.view.InputEventReceiver.consumeBatchedInputEvents(InputEventReceiver.java:177)
        at android.view.ViewRootImpl.doConsumeBatchedInput(ViewRootImpl.java:6735)
        at android.view.ViewRootImpl$ConsumeBatchedInputRunnable.run(ViewRootImpl.java:6787)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:911)
        at android.view.Choreographer.doCallbacks(Choreographer.java:723)
        at android.view.Choreographer.doFrame(Choreographer.java:652)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
        at android.os.Handler.handleCallback(Handler.java:790)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:440)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

</details>

### PR DESCRIPTION
**Pull request by charlag**
Fix problems with ID sorting, fix crashes with new [Pleroma flake IDs](https://git.pleroma.social/pleroma/pleroma/merge_requests/645)

(I see that it conflicts with master)

FIxes #1013

### CODE REVIEW

**Comment by charlag** [comment]
Oh, I'm not sure I did everything correctly, I branched from 4.3 initially...

**Comment by charlag** [comment]
Need to make sure that fe535c2 is in the current branch

**Comment by connyduck** [comment]
I just confirmed with the original report that Tusky 4.1 does not crash with flakeids. So we don't need a hotfix, sorry for the confusion. Can you send a new PR branched from master?

**Comment by charlag** [comment]
Yup, should be able to cherry pick or something, no problem, will do in the evening.

File: app/src/main/java/com/keylesspalace/tusky/util/StringUtils.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        if (builder[i] > '0') {
+            builder[i].dec()
+            break
+        }
+        i++
+    }
+    // All characters were '0'
+    if (i == builder.lastIndex && this.isNotEmpty()) {
+        // Remove one character
+        return this.substring(1)
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/util/StringUtils.kt]: Wait, this can be a bug
1000 - 1 = 999

**Comment by charlag** [comment]
I pushed completely broken `String.dec()` implementation first, oops, should sleep more and write more tests I guess

**Pull request merged**